File: src/characters/baron.ml
open! Core
open! Import

include Character.Make (struct
    type t =
      { kind : Kind.t
      ; alignment : Alignment.t
      }

    let id = "baron"
    let name = "Baron"
    let t = { kind = Kind.Minion; alignment = Alignment.Evil }
    let kind { kind; _ } = kind
    let alignment { alignment; _ } = alignment
  end)

let night_action ~player_id:_ ~night:_ = None
let day_action ~player_id:_ = None

let on_setup ~player_id:_ =
  Some (Botc_exec.log "Baron is in play: replace 2 Townsfolk with Outsiders")
;;

let on_nominated ~player_id:_ ~nominator:_ = None
let on_executed ~player_id:_ = None
let on_night_kill ~player_id:_ = None

File: src/characters/baron.rs
//! Baron (Minion): adds two Outsiders to the game. This is a bag-composition
//! effect resolved before seating, so at runtime the Baron is inert — the extra
//! Outsiders are already present in the roster the game was built with.

use crate::ability::Ability;
use crate::ctx::Ctx;
use crate::ids::PlayerId;
use crate::role::{Alignment, CharacterInfo, Kind};

character!(Baron, "baron", "Baron", Kind::Minion, Alignment::Evil);

impl Ability for Baron {
    fn info(&self) -> CharacterInfo {
        Self::INFO
    }

    fn on_setup(&self, ctx: &mut Ctx, _me: PlayerId) {
        ctx.log("Baron in play: the game was built with two extra Outsiders");
    }
}
